OpenRA.Mods.Common/Traits/Upgrades/UpgradeManager.cs

```diff
@@ -19,7 +19,7 @@
 namespace OpenRA.Mods.Common.Traits
 {
 	[Desc("Attach this to a unit to enable dynamic upgrades by warheads, experience, crates, support powers, etc.")]
-	public class UpgradeManagerInfo : ITraitInfo, Requires<IUpgradableInfo>
+	public class UpgradeManagerInfo : ITraitInfo, InitializeAfter<IUpgradableInfo>
```

Comment:
This might cause us dependency loops in the future (if we wanted to e.g. make `DeployToUpgrade` itself upgradeable).  How about we use `INotifyCreated` instead of the Lazy?